### PR TITLE
Data processing

### PR DESCRIPTION
**Pull request by matovu-farid**
In this branch, the following changes were made:
- The models were setup 
- The associations between the models were added
- The custom methods were added to the models

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class Like < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
   belongs_to :post
+  def update_post_likes_counter(value)
+    post = Post.where('id = ?', post_id).first
+    post.likes_counter = value
+    post.save
+  end
```
Comment (medaminedev66) [inline app/models/like.rb:8]: - [ ] A faster way, would be:
```rb
def update_likes_counter
  post.increment!(:likes_counter)
end
```

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,4 +1,9 @@
 class Comment < ApplicationRecord
-  belongs_to :user
   belongs_to :post
+  belongs_to :author, class_name: 'User'
+  def update_post_comment_counter(value)
+    post = Post.where('id = ?', post_id).first
+    post.comments_counter = value
+    post.save
+  end
```
Comment (medaminedev66) [inline app/models/comment.rb:8]: - [ ] A faster way (performance wise) would be
```rb
def update_comments_counter
  post.increment!(:comments_counter)
end
```

File: app/models/user.rb [hunk 3]
```diff
@@ -1,3 +1,9 @@
 class User < ApplicationRecord
-  has_many :posts, :likes, :comments
+  has_many :posts
+  has_many :likes
+  has_many :comments
```
Comment (medaminedev66) [inline app/models/user.rb:4]: - [ ]  According to the diagram, you should add the foreign key `author_id` to posts, comments, and likes:
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @matovu-farid  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- There are no linter errors ✔️
- You have a descriptive PR ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- The screenshot in your Readme is broken. I would recommend fixing that so your project is documented professionally.
- Kindly update the demo link to the online version link of `blog-app` rather than `covid metrics`.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
Hi @matovu-farid ,

Great job for implementing the changes proposed by the previous reviewer. In my view, your project is complete!   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :tada: :champagne: 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**